platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -1471,6 +1471,12 @@ - (NSExpression *)mgl_expressionLocalizedIntoLocale:(nullable NSLocale *)locale
                 if (localizedValues != self.constantValue) {
                     return [NSExpression expressionForConstantValue:localizedValues];
                 }
+            } else if ([self.constantValue isKindOfClass:[MGLAttributedExpression class]]) {
```

Comment:
What’s our theory for how this caused #14393? I was able to reproduce the issue in the standard Streets v10 and v11 styles. Is it because the `MGLSymbolStyleLayer.text` setter always returns an attributed expression now, even when the style JSON contains only a constant string value? If so, this may be indicative of a backwards-incompatible change that took place with the introduction of formatted expressions: a developer might reasonably expect to get an ordinary string out of the `text` property, just as we apparently did.